src/compiler/postprocess.rs

```diff
@@ -0,0 +1,101 @@
+use std::{collections::HashMap, io::{BufWriter, Write}, path::PathBuf, process::Command, sync::{Arc, Mutex, MutexGuard}};
+
+use itertools::Itertools;
+use wildmatch::WildMatchPattern;
+
+use crate::Cli;
+
+#[derive(Debug, Clone)]
+pub struct PostProcessor {
+    pub name: String,
+    pub bin: PathBuf,
+
+    command: Arc<Mutex<Command>>
```

Comment:
Is it necessary to wrap it in `Arc<Mutex<_>>`? I think that we use the `Command` in a single threaded manner

Reply:
it can't be shared mutably without it. `Command` is not a usual type, you either dance with lifetimes, or do this. im pretty sure there is a cleaner way so @Ph0enixKM push the commit or suggestion if you know it

Reply:
You don't need `Arc` (or "atomic reference count") because Amber is not multithreaded.  However, you do need a runtime borrow, and `RefCell` does this:
```
#[derive(Debug, Clone)]
pub struct PostProcessor {
    pub name: String,
    pub bin: PathBuf,
    command: Rc<RefCell<Command>>,
}
```
You can use it like this:
```
let mut spawned = self.cmd().borrow_mut().spawn()?;
```

Reply:
Pushed a new commit to fix this.